plugins/statuslines/lualine.nix

```diff
@@ -20,27 +20,35 @@ let
     });
     default = null;
   };
-  component_options = mode:
+  component_options =
     mkOption {
-      type = types.nullOr (types.submodule {
-        options = {
-          mode = mkOption {
-            type = types.str;
-            default = "${mode}";
-          };
-          icons_enabled = mkOption {
-            type = types.enum [ "True" "False" ];
-            default = "True";
-            description = "displays icons in alongside component";
-          };
-          icon = mkOption {
-            type = types.nullOr types.str;
-            default = null;
-            description = "displays icon in front of the component";
+      type = types.nullOr (types.listOf (types.oneOf [
+        types.str
+        (types.submodule {
+          options = {
+            name = mkOption {
```

Comment:
We probably want to keep the default name/mode that were used before so that it keep working for people who do not configure lualine in details.

Reply:
'mode' is just the name of a builtin component. I believe this is misleading.

I changed the name because no existing config could ever rely on the old implementation. Anyone setting the 'mode' field will simply find lualine failing. So I changed its misleading name.

But sure it's good to keep backward compatibility! What about adding a 'mode' alias to the 'name' field?

Reply:
Sorry I wasn't clear, my point wasn't about wether it is named name or mode. The problem I was raising is that it had a default value before, so the user that doesn't care much doesn't have to set it. Changing the name of the option is fine from my point of view.

Reply:
Ok I get your point. I'm adding it back.